src/main/java/com/ms3_inc/camel/extensions/rest/SwaggerRequestValidator.java

```
@@ -16,91 +16,143 @@
+import org.openapi4j.core.validation.ValidationException;
+import org.openapi4j.core.validation.ValidationResults;
+import org.openapi4j.operation.validator.model.Request;
+import org.openapi4j.operation.validator.model.impl.Body;
+import org.openapi4j.operation.validator.model.impl.DefaultRequest;
+import org.openapi4j.operation.validator.validation.RequestValidator;
+import org.openapi4j.parser.OpenApi3Parser;
+import org.openapi4j.parser.model.v3.OpenApi3;
+import org.openapi4j.parser.model.v3.Server;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.util.MultiValueMap;
-import org.springframework.web.util.UriComponentsBuilder;
+import org.springframework.core.io.ClassPathResource;
 
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
 public class SwaggerRequestValidator extends AsyncProcessorSupport {
-    private final Logger LOGGER = LoggerFactory.getLogger(getClass());
-    private final OpenApiInteractionValidator validator;
+    private final Logger LOGGER = LoggerFactory.getLogger(SwaggerRequestValidator.class);
+    private final RequestValidator openapi4jValidator;
+    private OpenApi3 api;
 
     public SwaggerRequestValidator(String specPath) {
-        validator = OpenApiInteractionValidator
-                .createFor(specPath)
-                .build();
+        api = null;
+
+        try {
+            api = new OpenApi3Parser()
+                    .parse(new ClassPathResource(specPath).getFile(), false);
+        } catch (ResolutionException re) {
+            LOGGER.error(re.getMessage());
+        } catch (ValidationException ve) {
+            LOGGER.error(ve.getMessage());
+        } catch (IOException ioe) {
+            LOGGER.error(ioe.getMessage());
+        }
+
+        openapi4jValidator = new RequestValidator(api);
+    }
+
+    public SwaggerRequestValidator(String specPath, String basePath) {
+        api = null;
+
+        try {
```

We shouldn't just suppress these errors. If the parser can't resolve the API then we should fail. ie remove this try/catch

---

so just do `throws IOException, ResolutionException, ValidationException` on the constructors?

---

catch any of these 3 and rethrow it as an `new IllegalArgumentException(caughtExc)`